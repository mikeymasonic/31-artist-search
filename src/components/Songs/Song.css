@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300&display=swap');

.song{
  height: 3vw;
  width: 12vw;
  display: flex;
  justify-content: center;
  text-align: center;
  margin: 0 3vw 3vw 0;
  padding: 1vw;
  border: .2vw solid rgb(133, 220, 226);
  border-radius: 1vw;
  font-size: 1.1vw;
  font-family: 'Montserrat', sans-serif;
  font-weight: 200;
}

.songlink {
  text-decoration: none;
  color: #f7d89e;
}

.songlink:hover {
  color: white;
}

.song:hover {
  background: hotpink;
}
