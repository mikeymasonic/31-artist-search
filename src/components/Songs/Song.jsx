import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Song = ({ songTitle, artistName }) => {
  return (
    <>
      <Link to={`/lyrics/${artistName}/${songTitle}`}>
        <h3>{songTitle}</h3>
      </Link>
    </>
  );
};

Song.propTypes = {
  songTitle: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired
};

export default Song;
