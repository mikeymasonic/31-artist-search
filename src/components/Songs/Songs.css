.songs {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  margin-left: 3vw;
}

.songs li {
  list-style: none;
}