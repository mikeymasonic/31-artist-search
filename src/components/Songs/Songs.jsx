import React from 'react';
import PropTypes from 'prop-types';
import Song from './Song';

const Songs = ({ songsArray, artistName }) => {
  const songList = songsArray.map(({ songTitle, songId }) => (
    <li key={songId}>
      <Song
        songTitle={songTitle}
        artistName={artistName}
      />
    </li>
  ));

  return (
    <ul>
      {songList}
    </ul>
  );
};

Songs.propTypes = {
  songsArray: PropTypes.arrayOf(PropTypes.shape({
    songTitle: PropTypes.string.isRequired,
    songId: PropTypes.string
  })).isRequired,
  artistName: PropTypes.string.isRequired
};

export default Songs;
