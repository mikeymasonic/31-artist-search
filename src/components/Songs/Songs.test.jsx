import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs component', () => {
  it('renders Songs', () =>{
    const wrapper = shallow(<Songs artistName='Silkworm' songsArray={[{
      songId: 'laskd311klasjd', songTitle: 'Never Met A Man I Didn\'t Like' }, {
      songId: 'laskd311klasjd', songTitle: 'Never Met A Man I Didn\'t Like' } 
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
