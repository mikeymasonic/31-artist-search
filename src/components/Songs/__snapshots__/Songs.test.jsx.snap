// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders Songs 1`] = `
<ul
  className="songs"
>
  <li
    key="laskd311klasjd"
  >
    <Song
      artistName="Silkworm"
      songTitle="Never Met A Man I Didn't Like"
    />
  </li>
  <li
    key="laskd311klasjd"
  >
    <Song
      artistName="Silkworm"
      songTitle="Never Met A Man I Didn't Like"
    />
  </li>
</ul>
`;
